compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs

```diff
@@ -745,14 +745,20 @@ fn shrink_file(
 ) -> Option<(Span, String, usize)> {
     let lo_byte = spans.iter().map(|s| s.lo()).min()?;
     let lo_loc = sm.lookup_char_pos(lo_byte);
-    let lo = lo_loc.file.line_bounds(lo_loc.line.saturating_sub(1)).start;
 
     let hi_byte = spans.iter().map(|s| s.hi()).max()?;
     let hi_loc = sm.lookup_char_pos(hi_byte);
+
+    if lo_loc.file.name != hi_loc.file.name {
```

Comment:
```suggestion
    if lo_loc.file.stable_id != hi_loc.file.stable_id {
```
Using the `stable_id` [avoid issues with file name collisions when multiple versions of the same crate are present in the dependency graph](https://github.com/rust-lang/rust/blob/86b95ebc24092acac75e205d95e84e6d4539601f/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs#L725-L728). In theory, [`collect_annotations`](https://github.com/rust-lang/rust/blob/86b95ebc24092acac75e205d95e84e6d4539601f/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs#L698) should be ensuring that the `stable_id`s match but it would be better to have it checked in both places.

Reply:
Doing cleanup for the other places in another PR makes sense. I can open a PR for that in the next few days if you'd like.